File: app/src/main/java/com/gnoemes/shikimoriapp/presentation/presenter/search/SearchPaginator.java
package com.gnoemes.shikimoriapp.presentation.presenter.search;

import com.gnoemes.shikimoriapp.entity.search.domain.FilterItem;
import com.gnoemes.shikimoriapp.presentation.presenter.common.Paginator;

import java.util.HashMap;
import java.util.List;

public interface SearchPaginator extends Paginator {

    void setFilters(HashMap<String, List<FilterItem>> filters);

    boolean isFirstPage();
}

File: app/src/main/java/com/gnoemes/shikimoriapp/presentation/presenter/search/SearchPaginator.kt
package com.gnoemes.shikimoriapp.presentation.presenter.search

import com.gnoemes.shikimoriapp.entity.app.domain.Type
import com.gnoemes.shikimoriapp.entity.search.domain.FilterItem
import com.gnoemes.shikimoriapp.presentation.presenter.common.Paginator

interface SearchPaginator : Paginator {
    fun setFilters(filters: HashMap<String, MutableList<FilterItem>>)
    fun setType(type: Type)
    fun isFirstPage(): Boolean
}